### PR TITLE
Scroll bar added

### PR DESCRIPTION
**Pull request by Souvik34**
Scroll bar added

Issue No: #299

### CODE REVIEW

**Comment by Souvik34** [comment]
Scroll bar added

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "react-particles": "^2.12.2",
     "react-responsive-carousel": "^3.2.23",
     "react-router-dom": "^6.16.0",
+    "react-scrollbars-custom": "^4.1.1",
```
Comment (harshkhandeparkar) [inline package.json]: Is this used anywhere?

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -7,14 +7,20 @@ import kwoc_logo from "../assets/kwoc_logo.png";
 import { ROUTER_PATHS, GH_OAUTH_URL } from "../util/constants";
 import { useAuthContext } from "../util/auth";
 import { UserType } from "../util/types";
+import "./Scrollbar.css";
 
+
+
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: Please remove unnecessary changes.

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -96,6 +104,13 @@ function LoginButton({ isMobile }: { isMobile: boolean }) {
 
 function Navbar() {
   const [mobileMenuOpen, setMobileMenuOpen] = useState(false);
+  const [bodyStyle, setBodyStyle] = useState({});
+
+  useEffect(() => {
+    setBodyStyle({
+      overflow: "hidden",
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: Is the useEffect necessary?

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -139,11 +155,20 @@ function Navbar() {
...
   );
 }
 
 function MobileNavbar({ toggleMobileMenu }: { toggleMobileMenu: () => void }) {
+  const [bodyStyle, setBodyStyle] = useState({});
+
+  useEffect(() => {
+    setBodyStyle({
+      overflow: "hidden",
+    });
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: Same here, seems like it does nothing.

File: src/components/Scrollbar.css
Comment (harshkhandeparkar) [inline src/components/Scrollbar.css:1]: Please follow the project structure. Add this code to `main.css`.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

**Comment by harshkhandeparkar** [comment]
@Souvik34 run `pnpm build` locally and fix any errors it displays.

File: src/components/Scrollbar.css
Comment (Souvik34) [reply src/components/Scrollbar.css:1]: Updated the code!

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -15,6 +15,7 @@ const LINKS = [
   { name: "FAQs", link: ROUTER_PATHS.FAQ },
 ];
 
+
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: Remove these as well

File: src/components/Navbar.tsx (hunk 3 above)
Comment (harshkhandeparkar) [reply src/components/Navbar.tsx]: ?

File: src/components/Navbar.tsx [hunk 6]
```diff
@@ -106,6 +115,7 @@ function Navbar() {
   };
 
   return (
+    <div className="App" style={bodyStyle}>
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: Why is this div added here?

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: src/components/Navbar.tsx (hunk 6 above)
Comment (Souvik34) [reply src/components/Navbar.tsx]: Actually for applying the CSS, to the whole webpage. Then I integrated the css code to  main.css. I removed this portion  now. 

I think now it is perfect.

File: src/components/Navbar.tsx [hunk 7]
```diff
@@ -143,7 +144,9 @@ function Navbar() {
 }
 
 function MobileNavbar({ toggleMobileMenu }: { toggleMobileMenu: () => void }) {
+  const [bodyStyle] = useState({});
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: It's still there.

File: src/components/Navbar.tsx [hunk 8]
```diff
@@ -106,6 +106,7 @@ function Navbar() {
   };
 
   return (
+
```
Comment (harshkhandeparkar) [inline src/components/Navbar.tsx]: And revert these extra changes please.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

**Comment by harshkhandeparkar** [comment]
@Souvik34 Can you send a screenshot of how the scrollbar is supposed to look? Also check if it works on Firefox (I don't think it is working for me).

**Comment by Souvik34** [comment]
image
It is looking like this in Brave

image
It is looking like this in Edge.

image
It is looking like this in Chrome.

**Comment by harshkhandeparkar** [comment]
It is working on Chromium based browsers. Is it possible to make it work on Firefox?

**Comment by Souvik34** [comment]
Alright I am working for it asap.

**Comment by harshkhandeparkar** [comment]
@Souvik34 any update?

**Comment by Souvik34** [comment]
Updated

File: src/main.css [hunk 9]
```diff
@@ -14,3 +14,29 @@
 body {
   background: url("./assets/stars.png"), #0a0a19;
 }
+ /* ===== Scrollbar CSS ===== */
+   * {
+    scrollbar-width: thin;
```
Comment (harshkhandeparkar) [inline src/main.css]: Can you make both (firefox and chromium) scrollbars the same width? The Firefox one is too small. In fact it would be good if they are both changed to 5px.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

**Review by harshkhandeparkar — APPROVED** [review]

**Comment by harshkhandeparkar** [comment]
Thanks @Souvik34! :rocket:
